Этап 2. Main service.

KanID92 opened the pull request.

KanID92 committed:
add:
- API for work with users: /admin/users;

fix:
- StatClient. add try-catch for exceptions

KanID92 committed:
add:
- API for work with users: /admin/users;

fix:
- StatClient. add try-catch for exceptions

KanID92 committed:
add:
- API for work with users: /admin/users;

fix:
- StatClient. add try-catch for exceptions

KanID92 committed:
Merge remote-tracking branch 'origin/main_svc_admin_users' into main_svc_admin_users

KanID92 committed:
- fix port

KanID92 committed:
- fix pom ewm-service;

KanID92 committed:
- fix v3;

KanID92 committed:
- fix v4;

KanID92 committed:
Add:
- EventController;
- EventDto's;
- Event model;

KanID92 committed:
- fix v5;

KseniaYakovenko committed:
categories

KseniaYakovenko committed:
categories

KanID92 committed:
Merge pull request #6 from KanID92/main_svc_admin_users

KanID92 committed:
Add:
- EventService(user) without statistic;
- EventMapper;
- LocationRepository;
- ConflictException;
- AccessException;

KseniaYakovenko committed:
categories

KanID92 committed:
Merge branch 'refs/heads/main_svc' into main_svc_events

KanID92 committed:
Merge branch 'refs/heads/main_svc_categories_rebase' into main_svc_events

# Conflicts:
#	ewm-service/src/main/java/ru/practicum/ewm/dto/category/CategoryDto.java
#	ewm-service/src/main/java/ru/practicum/ewm/dto/event/NewEventDto.java
#	ewm-service/src/main/java/ru/practicum/ewm/entity/Category.java
#	ewm-service/src/main/java/ru/practicum/ewm/handler/ErrorHandler.java
#	ewm-service/src/main/java/ru/practicum/ewm/mapper/EventMapper.java
#	ewm-service/src/main/java/ru/practicum/ewm/repository/EventRepository.java
#	ewm-service/src/main/java/ru/practicum/ewm/service/EventServiceImpl.java
#	ewm-service/src/main/resources/schema.sql

KanID92 committed:
Merge pull request #8 from KanID92/main_svc_categories_rebase

Main svc categories rebase

KanID92 committed:
fix:
- EventMapper;
- EventServiceImpl;

KanID92 committed:
- Events: interim fix and adding;

KanID92 committed:
Add:
- added events without requests;

KanID92 committed:
Merge pull request #9 from KanID92/main_svc_events

KseniaYakovenko committed:
compilation

KseniaYakovenko committed:
compilation

KanID92 committed:
Merge pull request #10 from KanID92/main_svc_compilations

compilation

KanID92 committed:
- Requests (private api, schema, model, dto, mapper).

KanID92 committed:
Add:
- requests;
fix:
- by postman test (not all);

KseniaYakovenko committed:
fix request

KseniaYakovenko committed:
Merge branch 'main_svc_requests' into main_svc

KseniaYakovenko committed:
fix all

KanID92 committed:
Merge remote-tracking branch 'refs/remotes/origin/main_svc_compilations' into main_svc_requests

KseniaYakovenko committed:
fix all

KanID92 committed:
Merge remote-tracking branch 'refs/remotes/origin/main_svc_all_fix' into main_svc_requests

KanID92 committed:
- fix some tests.

KanID92 committed:
- fix some tests (except 2).

KseniaYakovenko committed:
fix all

KanID92 committed:
- fix checkstyle

KanID92 committed:
- fix comparing RequestServiceImpl;

KanID92 committed:
- fix comparing RequestServiceImpl (v.2);

kesch9 commented on ewm-service/src/main/java/ru/practicum/ewm/entity/Event.java:
```diff
@@ -0,0 +1,73 @@
+@Setter
+@EqualsAndHashCode(of = {"id"})
+@Table(name = "EVENTS")
+@ToString
```
Обязательно
Использовать toString также не рекомендуется с entity
https://habr.com/ru/companies/haulmont/articles/564682/

KanID92 replied:
Откорректировано.

kesch9 commented on ewm-service/src/main/java/ru/practicum/ewm/service/EventServiceImpl.java:
```diff
@@ -0,0 +1,321 @@
+
+    @Override
+    @Transactional(readOnly = true)
+    public List<EventShortDto> getAllByInitiatorOrPublic(EventSearchParams searchParams, HitDto hitDto) {
```
Обязательно
Разделите метод на два метода, для обработки публичного и приватного endpoint. В вашей реализации две ветки и нет общей логики, поэтому лучше разнести

KanID92 replied:
Откорректировано

kesch9 commented on ewm-service/src/main/java/ru/practicum/ewm/service/EventServiceImpl.java:
```diff
@@ -0,0 +1,321 @@
+                rangeEnd = rangeStart.plusYears(100);
+            }
+
+//            if (!publicSearchParams.getOnlyAvailable()) { //TODO after Requests
```
Обязательно
На ревью не должно быть закоментированного кода, либо доделайте, либо удалите

KanID92 replied:
Откорректировано.

kesch9 commented on ewm-service/src/main/java/ru/practicum/ewm/service/EventServiceImpl.java:
```diff
@@ -0,0 +1,321 @@
+
+            eventListBySearch = eventRepository.findAll(booleanExpression, page).stream().toList();
+
+            statClient.saveHit(hitDto); //TODO перенести?
```
Обязательно
Закрыть TODO

KanID92 replied:
Откорректировано

kesch9 commented on ewm-service/src/main/java/ru/practicum/ewm/controller/publ/PublicEventController.java:
```diff
@@ -0,0 +1,101 @@
+        List<EventShortDto> eventShortDtoList = eventService.getAllByInitiatorOrPublic(eventSearchParams, hitDto);
+        log.info("<== GET /events Returning public searching events. List size: {}",
+                eventShortDtoList.size());
+        //TODO Statistic?
```
Обязательно
Закрыть TODO

KanID92 replied:
Откорректировано

kesch9 commented on ewm-service/src/main/java/ru/practicum/ewm/repository/EventRepository.java:
```diff
@@ -0,0 +1,20 @@
+
+    Optional<Event> findByInitiatorIdAndId(long initiatorId, long eventId);
+
+//    List<Event> findAll(BooleanExpression booleanExpression, Pageable pageable);
```
Обязательно
На ревью не должно быть закоментированного кода, либо доделайте, либо удалите

KanID92 replied:
Откорректировано.

kesch9 commented on ewm-service/src/main/java/ru/practicum/ewm/service/EventServiceImpl.java:
```diff
@@ -0,0 +1,321 @@
+
+        return eventRepository.findAll(booleanExpression, page)
+                .stream()
+                .peek(event -> event.setConfirmedRequests(
```
Обязательно
Не рекомендуется использовать peek для бизнес логики, обычно используется для отладки кода
https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#peek-java.util.function.Consumer-
https://4comprehension.com/idiomatic-peeking/
https://blog.jooq.org/java-8-friday-10-subtle-mistakes-when-using-the-streams-api/
Нужно переписать заполнения request

KanID92 replied:
Откорректировано

kesch9 commented on ewm-service/src/main/java/ru/practicum/ewm/service/RequestServiceImpl.java:
```diff
@@ -0,0 +1,171 @@
+@RequiredArgsConstructor
+public class RequestServiceImpl implements RequestService {
+
+    private static final Logger log = LoggerFactory.getLogger(RequestServiceImpl.class);
```
Обязательно
log не используется надо удалить и для создания используйте ломбок @Slf4j

KanID92 replied:
Откорректировано.

kesch9 commented on ewm-service/src/main/java/ru/practicum/ewm/service/RequestServiceImpl.java:
```diff
@@ -0,0 +1,171 @@
+                        status, request.getId());
+                Request modifiedRequest = requestRepository.findById(request.getId())
+                        .orElseThrow(() -> new NotFoundException("Request with id " + request.getId() + " not found"));
+                System.out.println("Обновленный " + modifiedRequest.getId() + " " + modifiedRequest.getStatus());
```
Обязательно
Нужно использовать log вместо System.out.println

kesch9 replied:
Проверьте код и вместо System.out.println используйте log в остальных местах.

KanID92 replied:
откорректировано

kesch9 reviewed (commented):
Хорошо поработали над заданием нужно немного доработать

KanID92 committed:
fix according to the review:
- delete TODO and log by System.out.println;
- delete @toString in entity;
- EventServiceImpl. Separate one method into two (getAllByInitiator(EventSearchParams searchParams), getAllByPublic(EventSearchParams searchParams, HitDto hitDto);
- EventServiceImpl.getAllByAdmin. Remove ".peek" from stream.

kesch9 reviewed (approved):
Approve
